platform/security/idp/security-idp-server/src/main/java/org/codice/ddf/security/idp/server/IdpEndpoint.java

```diff
@@ -0,0 +1,820 @@
+        String assertionConsumerServiceURL = null;
+        LOGGER.debug("Attempting to determine AssertionConsumerServiceURL.");
+        //if the AuthnRequest specifies a URL, use that
+        if (authnRequest.getAssertionConsumerServiceURL() != null) {
+            LOGGER.debug("Using AssertionConsumerServiceURL from AuthnRequest: {}", authnRequest.getAssertionConsumerServiceURL());
+            assertionConsumerServiceURL = authnRequest.getAssertionConsumerServiceURL();
+        } else {
+            //check metadata
+            EntityDescriptor entityDescriptor = serviceProviders
+                    .get(authnRequest.getIssuer().getValue());
+            SPSSODescriptor spssoDescriptor = entityDescriptor
+                    .getSPSSODescriptor(SamlProtocol.SUPPORTED_PROTOCOL);
+            AssertionConsumerService defaultAssertionConsumerService = spssoDescriptor
+                    .getDefaultAssertionConsumerService();
+            //see if the default service uses our supported bindings, and then use that
+            if (defaultAssertionConsumerService.getBinding().equals(HTTP_POST_BINDING)
+                    || defaultAssertionConsumerService.getBinding().equals(HTTP_REDIRECT_BINDING)) {
+                LOGGER.debug("Using AssertionConsumerServiceURL from default assertion consumer service: {}", defaultAssertionConsumerService.getLocation());
+                assertionConsumerServiceURL = defaultAssertionConsumerService.getLocation();
+            } else {
+                //if default doesn't work, check any others that are defined and use the first one that supports our bindings
+                for (AssertionConsumerService assertionConsumerService : spssoDescriptor
+                        .getAssertionConsumerServices()) {
+                    if (assertionConsumerService.getBinding().equals(HTTP_POST_BINDING)
+                            || assertionConsumerService.getBinding()
+                            .equals(HTTP_REDIRECT_BINDING)) {
+                        LOGGER.debug("Using AssertionConsumerServiceURL from supported binding: {}", assertionConsumerService.getLocation());
+                        assertionConsumerServiceURL = assertionConsumerService.getLocation();
+                        break;
+                    }
+                }
+            }
+
+        }
+        if (assertionConsumerServiceURL == null) {
+            throw new IllegalArgumentException("No valid AssertionConsumerServiceURL available for given AuthnRequest.");
+        }
+        return assertionConsumerServiceURL;
+    }
+
+    protected Crypto getSignatureCrypto() {
+        if (signatureCrypto == null && signaturePropertiesPath != null) {
+            Properties sigProperties = PropertiesLoader.loadProperties(signaturePropertiesPath);
+            if (sigProperties != null) {
+                try {
+                    PasswordEncryptor passwordEncryptor = new PasswordEncryptor() {
```

Comment:
Instead of creating these on every request, could we just maintain a single instance that is created/updated on calls to `setEncryptionService()`?